java/fury-core/src/main/java/org/apache/fury/pool/ClassLoaderFuryPooled.java

```diff
@@ -122,6 +124,20 @@ private void addFury() {
   }
 
   void setFactoryCallback(Consumer<Fury> factoryCallback) {
```

Comment:
This method should not be called concurently, users should invoke regitster and other fury config methods when creating ThreadSafeFury. It's not necessary to add a lock

Reply:
In fact, all our register and config functions should be called when `ThreadSafeFury` is being created, but the purpose of adding the `processCallback` function should be contrary to this design. The `processCallback` interface allows register or config operations to be performed after created `ThreadSafeFury`.

Reply:
```java
ThreadSafeFury safeFury = Fury.builder()
    .requireClassRegistration(true)
    .buildThreadSafeFury();
safeFury.register(A.class);
safeFury.register(B.class);

// Thread A
new Thread(() -> {
  safeFury.serialize(new A());
}).start();

// Thread B
new Thread(() -> {
  // processCallback(fury -> fury.register(clz));
  safeFury.register(C.class);
  safeFury.serialize(new C());
}).start();
```

As shown in the above Demo, how can we avoid `safeFury.register(C.class)` being called in Thread B.

Reply:
Then we should add code to avoid this, such as capture Thread, and check thread in later call

Reply:
I think this is a feasible approach, but I think the methods that implement the ThreadSafeFury interface should be thread-safe, and all methods in ThreadSafeFury should be allowed to be called in any thread.